bgpd/bgp_fsm.c

```diff
@@ -1935,6 +1935,7 @@ static int bgp_establish(struct peer *peer)
 	hash_release(peer->bgp->peerhash, peer);
 	hash_get(peer->bgp->peerhash, peer, hash_alloc_intern);
 
+	bgp_bfd_deregister_peer(peer);
```

Comment:
Is this something to handle BFD UP notification directly when BGP establishes again after it flaps? Not sure If I understood problem properly with commit message.

Reply:
local BFD thinks it is up, while remote BFD thinks it is down and should go to init state.
but local BFD receives down, and sees this as a DOWN event.

not good.
so the idea consists in resetting bfd from bgp once bgp gets established.
there is no reset API, but this is the idea.